great_expectations/rule_based_profiler/parameter_builder/parameter_builder.py

```diff
@@ -553,6 +553,46 @@ def _sanitize_metric_computation(
 
         return metric_values
 
+    @staticmethod
+    def _get_best_candidate_above_threshold(
+        candidate_ratio_dict: Dict[str, float],
+        threshold: float,
+    ) -> Tuple[Optional[str], float]:
+        """
+        Helper method to calculate which candidate strings or patterns are the best match (ie. highest ratio),
+        provided they are also above the threshold.
+        """
+        best_candidate: Optional[str] = None
+        best_ratio: float = 0.0
+
+        candidate: str
+        ratio: float
+        for candidate, ratio in candidate_ratio_dict.items():
+            if ratio > best_ratio and ratio >= threshold:
+                best_candidate = candidate
+                best_ratio = ratio
+
+        return best_candidate, best_ratio
+
+    @staticmethod
+    def _get_sorted_candidates_and_ratios(
+        candidate_ratio_dict: Dict[str, float],
+    ) -> Tuple[List[float], List[str]]:
+        """
+        Helper method to sort all candidate strings strings or patterns by success ratio (how well they matched the domain).
+
+        Returns Tuple(ratio, sorted_strings)
+        """
+        strings: List[str] = list(candidate_ratio_dict.keys())
+        ratios: List[float] = list(candidate_ratio_dict.values())
+
+        sorted_strings: List[str] = [
+            string for ratio, string in sorted(zip(ratios, strings), reverse=True)
+        ]
+        ratios.sort(reverse=True)
+
+        return ratios, sorted_strings
+
```

Comment:
Method to calculate the best candidate and sorted dictionary of all candidates (for `details`) is now in the base `ParameterBuilder` class, and is shared by both the `datetime` and `regex` parameter builders